### PR TITLE
feat: Solution unable to get avatar from gravatar 

### LINKED ISSUE #188: Provide default avatar image like GitHub when the user doesn't set one and the gravatar.com is unaccessable
If the user doesn't set a custom avatar image, the Go backend (already implemented) will try to fetch avatar from gravatar.com. If gravatar.com is unaccessable (e.g. blocked in some areas), then the frontend will provide a default GitHub-identicon-style avatar image.

The default image is dynamically generated in frontend by this lib: https://github.com/stewartlord/identicon.js (can use username as hash for identicon).

Note: the default image is NOT stored in backend or DB.

Fix: https://forum.casbin.com/t/115

### PR DESCRIPTION
**Pull request by MRGUOKING**
Fix: https://github.com/casbin/casnode/issues/188

Signed-off-by: MRGUOKING <420919469@qq.com>

If there is no avatar in the database, the avatar will be generated directly according to the user name

Signed-off-by: MRGUOKING <420919469@qq.com>

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -56,6 +57,7 @@
       "last 1 safari version"
     ]
   },
+
```
Comment (hsluoyz) [inline web/package.json]: no new line

File: web/src/Setting.js [hunk 2]
```diff
@@ -126,8 +127,9 @@ export function getUserAvatar(username, isLarge = false) {
     return null;
   }
 
-  let gravatarStr = md5(username);
-  return `https://www.gravatar.com/avatar/${gravatarStr}?d=retro`;
+  let identicon = new Identicon(md5(username), 420);
+  return "data:image/png;base64," + identicon;
+  // return `https://www.gravatar.com/avatar/${gravatarStr}?d=retro`;
```
Comment (hsluoyz) [inline web/src/Setting.js]: remove all comments here

File: web/yarn.lock [hunk 3]
```diff
@@ -5834,6 +5834,11 @@ icss-utils@^4.0.0, icss-utils@^4.1.1:
   dependencies:
     postcss "^7.0.14"
 
+identicon.js@^2.3.3:
+  version "2.3.3"
+  resolved "https://registry.nlark.com/identicon.js/download/identicon.js-2.3.3.tgz#c505b8d60ecc6ea13bbd991a33964c44c1ad60a1"
```
Comment (hsluoyz) [inline web/yarn.lock]: use: registry.yarnpkg.com

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/Setting.js (hunk 2 above)
Comment (MRGUOKING) [reply web/src/Setting.js]: I hava solved the three problems you said before,thank you for your advices.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**